### ISSUE TITLE
config_uri parsing

### ISSUE THREAD

**Issue by huntcsg** [Pylons/plaster#1]
I've been thinking about how we would generically accept uri's in plaster to make it as adaptable as possible for the variety of loaders that could be written. At first I considered the idea of "hand parsing" them based on the typical case:
    `'development.ini' => scheme='ini', uri='development.ini', and section=None`
    `'development.ini#main' => scheme = 'ini', uri='development.ini', and section='main'`
etc.

But I'm now considering that perhaps we should use the standard library module urlparse:

```
>>> import urlparse
>>> urlparse.urlparse('development.ini')
ParseResult(scheme='', netloc='', path='development.ini', params='', query='', fragment='')
>>> urlparse.urlparse('development.ini#main')
ParseResult(scheme='', netloc='', path='development.ini', params='', query='', fragment='main')
>>> urlparse.urlparse('ini+pastedeploy://development.ini#myapp')
ParseResult(scheme='ini+pastedeploy', netloc='development.ini', path='', params='', query='', fragment='myapp')
>>> urlparse.urlparse('http://configuration_server_url.com:8888/development#main')
ParseResult(scheme='http', netloc='configuration_server_url.com:8888', path='/development', params='', query='', fragment='main')
```

It seems to handle each of the cases pretty well. Internally we could set some defaults, such as setting the scheme based on file extension, in absence of an explicit scheme. Part of the plaster specification would be that it would pass a `urlparse.ParseResult` object to the Loader.

Another idea was that if the Loader wanted to be able to stand alone and accept a string representation, that could be a class attribute/flag that could be looked up, and then plaster would just passthrough the string, after using urlparse.urlparse to get the scheme from the url.

**Comment by mmerickel** [comment]
I agree with all of this - my initial implementation was also using urlparse. I was planning to convert that to a `PlasterURL` object prior to passing it to the loaders but to use urlparse under the hood. There will be a public api `plaster.parse_url` that libraries could use if they wanted to work in a standalone way.

Things have been busy but hopefully I can push that this week.

**Comment by mmerickel** [comment]
I will point out though that SQLAlchemy surprisingly does not use `urlparse` but rather its own regex parser. I am not sure what the reasoning is here but it was interesting to me to discover. https://github.com/zzzeek/sqlalchemy/blob/master/lib/sqlalchemy/engine/url.py#L199

**Comment by mmerickel** [comment]
Shipped 0.1.0 using `urlparse`. Seems to work pretty well for now.
